Assets/Scripts/EnemyController.cs

```diff
@@ -104,8 +104,10 @@ void Update()
                 photonView.RPC("Blobify", RpcTarget.All);
             }
             
+
             if(distanceToKeyLocationToDespawn > Vector3.Distance(gameObject.FindClosestObject(keyLocations).transform.position, transform.position))
             {
+                gameObject.FindClosestObject(keyLocations).GetComponent<KeyLocationController>().LoseHealth();
                 PhotonNetwork.Destroy(gameObject);
```

Comment:
```suggestion
            GameObject closestKeyLocation = gameObject.FindClosestObject(keyLocations)
            if(distanceToKeyLocationToDespawn > Vector3.Distance(closestKeyLocation.transform.position, transform.position))
            {
                closestKeyLocation.GetComponent<KeyLocationController>().LoseHealth();
                PhotonNetwork.Destroy(gameObject);
```
Finding closest key location shouldn't be done twice with no specific reason. Better put it in a variable, so it's more efficient